### ISSUE TITLE
Servicedesk email sent before user makes vetting choice

### ISSUE THREAD

**Issue by phavekes** [OpenConext/Stepup-Project#391]
[This issue is imported from pivotal - Originaly created at Nov 17, 2022 by Peter Clijsters](https://www.pivotaltracker.com/story/show/183826628)

During token registration, if a users has a choice how to activate his token (other token or servicedesk), the "activate your token" email with servicedesk instructions is already sent **before** a choice is made.  This is a bug in the activation flow. This email should only be sent if the user choses the servicedesk activation method.

**Comment by phavekes** [comment]
This also happens when using the recovery method for activating a token. (Peter Havekes - Nov 18, 2022)

**Comment by phavekes** [comment]
The registration mail message is now processed based on the fact a `PossessionProvenAndVerified` event is recorded. And that is recorded once the token verification in SS is finished.

From a UI standpoint, after token verification we display the available vetting types. As the number of available vetting types depends on different conditions. That we can only determine after token verification finished.

In order to postpone the sending of the verification mail message we will need to let the Registration mail processor listen to another event. A new event that is not yet recorded. That event should be triggered once a vetting type is selected. That new vetting type event should be made to carry the activation code that is now part of the possession proven and verified event. (Michiel Kodde - Nov 21, 2022)

**Comment by phavekes** [comment]
We later decided to not send this mail message event based, as no relevant events are present to send this message on. So we will just have a new command that triggers sending the mail message. And not record any event from that command. (Michiel Kodde - Dec 8, 2022)
